download_and_install/compiling_from_source.md

```diff
@@ -0,0 +1,241 @@
+Compiling From Source
+=====================
+
+OLA source code lives on [GitHub](https://github.com/OpenLightingProject/ola).
+Stable releases are available from the [releases page](https://github.com/OpenLightingProject/ola/releases);
+for bleeding-edge development releases, run `git clone https://github.com/OpenLightingProject/ola.git`.
+
+Dependencies
+============
+
+You need a couple of libraries installed for everything to work correctly.
+
+First you’ll need at least the following:
+
+- [avahi](https://www.avahi.org) if you want to use network discovery.
+- [microhttpd](https://www.gnu.org/software/libmicrohttpd/) (if you want the web UI).
+- autoconf
+- automake
+- cppunit
+- ncurses
+- lex (or flex)
+- libftdi
+- liblo
+- libtool
+- libusb
+- pkg-config
+- [Protobuf](https://developers.google.com/protocol-buffers) (version 3.6 or later)
+- uuid or ossp uuid
+- yacc (or bison)
+- zlib
+
+The Python library also requires
+
+- Numpy
+- Protobuf for Python
+
+Debian/Ubuntu
+-------------
+
+Debian/Ubuntu users can install all dependencies with:
+
+    sudo apt-get install \
+        autoconf \
+        automake \
+        bison \
+        flex \
+        g++ \
+        libavahi-client-dev \
+        libcppunit-dev \
+        libftdi-dev \
+        liblo-dev \
+        libmicrohttpd-dev \
+        libncurses-dev \
+        libprotobuf-dev \
+        libprotoc-dev \
+        libtool \
+        libusb-1.0.0-dev \
+        make \
+        pkg-config \
+        uuid-dev \
+        zlib1g-dev
+
+CentOS/RHEL/Fedora
+------------------
+
+Distributions using RPM packages can install all dependencies with:
+
+    sudo dnf install \
+        autoconf \
+        automake \
+        avahi-devel \
+        bison \
+        cppunit-devel \
+        flex \
+        gcc-c++ \
+        libftdi-devel \
+        liblo-devel \
+        libmicrohttpd-devel \
+        libtool \
+        libusbx-devel \
+        make \
+        ncurses-devel \
+        protobuf-lite-devel \
+        uuid-devel \
+        zlib-devel
+
+macOS
+-----
+
+Dependencies are available from [MacPorts](https://www.macports.org/):
+
+    sudo port install \
+        autoconf \
+        automake \
+        bison \
+        cppunit \
+        flex \
+        libftdi1 \
+        liblo \
+        libmicrohttpd \
+        libtool \
+        libusb \
+        libuuid \
+        ncurses \
+        pkgconfig \
+        protobuf3-cpp \
+        zlib
+
+or [Homebrew](https://brew.sh/):
+
+    brew install \
+        autoconf \
+        automake \
+        bison \
+        cppunit \
+        flex \
+        libftdi \
+        liblo \
+        libmicrohttpd \
+        libtool \
+        libusb \
+        libuuid \
+        ncurses \
+        pkg-config \
+        protobuf \
+        zlib
+
+Python dependencies
+-------------------
+
+If you wish to build the Python library, additional libraries are required:
+
+    pip3 install --user \
+        numpy \
+        protobuf
```

Comment:
This seems to cause issue on some Ubuntu flavours (I think they end up in a different path and not found by configure), instead use:
```suggestion
    sudo apt-get install \
        python-numpy \
        python-protobuf
```

Or the python3 versions of them...

Reply:
Huh, I guess my local Ubuntu might be too new.  I'll check other distros for this issue - I was trying to keep Python with Python, so to speak, by using pip instead of the system package manager.

Reply:
Possibly, there's also the Python 2/3 thing, you can check some of the Travis runs and config history for what I had to do to make it work. This failure also matched what a user on IRC experienced.

Reply:
Alright, it looks like the Python ClientWrapperTest is failing on Ubuntu when using pip, but passing when using system packages.  It works either way on Fedora.  I'm baffled.  Anyway, updating the Python lib to note this problem.

Reply:
> Alright, it looks like the Python ClientWrapperTest is failing on Ubuntu when using pip, but passing when using system packages. It works either way on Fedora. I'm baffled. Anyway, updating the Python lib to note this problem.

Can you open an issue for this in OLA @danielskeenan as it would be good to get to the bottom of it. Some OS version and test log failure stuff would be useful too.

Reply:
Will do next time I flip back to Linux, where all of my test VMs are.

Reply:
Opened OpenLightingProject/ola#1714.  It turned out to be more interesting than I thought!